### DISCUSSION TITLE
How to implement a composite primary key

### DISCUSSION THREAD

**Discussion by jaohbib** [fastapi/sqlmodel#1244] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
class test(SQLModel, table=True):
     src_ppn: str = Field(primary_key=True)
     dst_ppn: str  = Field(primary_key=True)
   
     __table_args__ = (
         PrimaryKeyConstraint(src_ppn, dst_ppn),
     )
```

### Description

I want to create a table containing a composite primary key
I recive the following error 

ArgumentError: String column name or column expression for DDL constraint expected, got FieldInfo(annotation=NoneType, required=True).

### Operating System

Linux

### Operating System Details

Ubuntu 22.04

### SQLModel Version

0.0.22

### Python Version

3.9.7

**Comment by monchin** [comment]
I guess there's something wrong with `PrimaryKeyConstraint`. You csn try to remove it like [the example in the document](https://sqlmodel.tiangolo.com/tutorial/many-to-many/create-models-with-link/#link-table-model)

**Reply by acidnik** [reply]
in case someone landed here from the search engine, i think the problem could be fixed like this:
```python
     __table_args__ = (
         PrimaryKeyConstraint("src_ppn", "dst_ppn"), # note the quotes around fields
     )
```

**Comment by YuriiMotov** [comment, marked as answer]
You don't actually need `PrimaryKeyConstraint`.
Just specifying `primary_key=True` for both fields is enough to create composite primary key:

```py
from sqlalchemy import PrimaryKeyConstraint
from sqlmodel import SQLModel, Field, create_engine


class test(SQLModel, table=True):
    src_ppn: str = Field(primary_key=True)
    dst_ppn: str = Field(primary_key=True)

    __table_args__ = (PrimaryKeyConstraint("src_ppn", "dst_ppn"),)


class test2(SQLModel, table=True):
    src_ppn: str = Field(primary_key=True)
    dst_ppn: str = Field(primary_key=True)


engine = create_engine("sqlite:///", echo=True)


SQLModel.metadata.create_all(engine)

```

Output (as you can see there is no difference):
```
CREATE TABLE test (
        src_ppn VARCHAR NOT NULL, 
        dst_ppn VARCHAR NOT NULL, 
        PRIMARY KEY (src_ppn, dst_ppn)
)

CREATE TABLE test2 (
        src_ppn VARCHAR NOT NULL, 
        dst_ppn VARCHAR NOT NULL, 
        PRIMARY KEY (src_ppn, dst_ppn)
)
```
